### ISSUE TITLE
When the prompt exceeds the length

### ISSUE THREAD

**Issue by jsk1107** [microsoft/GLIP#61]
hello.

Q1.
The total number of classes in the custom data set I have is about 300. So it throws an error. I think this error occurs because the dimension of the logit is [:, :, 256] but the value index of my positive map is greater than 256. Am i right?

Q2.
In #37 "If the prompt exceeds the length, you can take a look at the inference codes about how we deal with the LVIS dataset (~1200 classes)" said.
Can converting coco annotations to LVIS annotations solve the error in Q1? If so, Is there an API that provides conversion between annotations?

**Comment by Haotian-Zhang** [comment]
@jsk1107 Thank you for the questions! Yes, the error is due to the maximum length of the text encoder input, which is 256. You can convert your label directly to LVIS format. There shall be no such big differences between the COCO and LVIS label files. The only thing you may do is modify the datasets section in the config file to make sure the model knows what kind of data you are specifying.

**Comment by jsk1107** [comment]
@Haotian-Zhang Problem solved. thank you

I have one more question. How should spaces be handled when entering prompts?

ex) aerosol can / aerosol_can
ex) air conditioner / air_conditioner

WordNet seems to provide underscores, not spaces. It is also underscore in LVIS comments. Is it a good practice to use underscores for correct prompting?

**Comment by Haotian-Zhang** [comment]
Hi @jsk1107, I don't think underscore make senses in our case. Since we are using BERT as our text encoder, and it was pre-trained on the natural sentences styles. Besides, our detection prompt is also using spaces in between the class categories. Please let us know if you have further concerns.

**Comment by jsk1107** [comment]
@Haotian-Zhang Thanks! I'll close this issue.

**Comment by nanfei666** [comment]
> @jsk1107 Thank you for the questions! Yes, the error is due to the maximum length of the text encoder input, which is 256. You can convert your label directly to LVIS format. There shall be no such big differences between the COCO and LVIS label files. The only thing you may do is modify the datasets section in the config file to make sure the model knows what kind of data you are specifying.

Hi, i am not sure which part should be modified in the config file if i have a coco formate annotation file while have much more class, could you point it out for me . thank you

**Comment by vishalvijay99** [comment]
@jsk1107 what changes did you make? Can you please elaborate
